// ngx-sidebar/bundles/ngx-sidebar.umd.js
(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/router'), require('@angular/platform-browser')) :
	typeof define === 'function' && define.amd ? define(['exports', '@angular/core', '@angular/router', '@angular/platform-browser'], factory) :
	(factory((global['ngx-sidebar'] = {}),global.ng.core,global.ng.router,global.ng.platformBrowser));
}(this, (function (exports,core,router,platformBrowser) { 'use strict';

var NgxSidebarComponent = /** @class */ (function () {
    function NgxSidebarComponent(renderer, router$$1, route) {
        this.renderer = renderer;
        this.router = router$$1;
        this.route = route;
        this.containerClass = 'main-ul';
        this.state = true;
    }
    NgxSidebarComponent.prototype.ngOnInit = function () {
    };
    NgxSidebarComponent.prototype.menuToggle = function (event, nav) {
        if (nav.hasOwnProperty('children')) {
            if (event.currentTarget.getAttribute('class') !== 'opened') {
                this.state = true;
            }
            this.renderer.setElementClass(event.currentTarget, 'opened', this.state);
            this.state = !this.state;
        }
    };
    return NgxSidebarComponent;
}());
NgxSidebarComponent.decorators = [
    { type: core.Component, args: [{
                selector: 'ngx-sidebar',
                template: "<ul [ngClass]='containerClass'>\n  <li *ngFor=\"let nav of navigation\"  [routerLinkActive]=\"nav.link ? 'opened' : ''\"  (click)=\"menuToggle($event, nav);$event.stopPropagation()\">\n    <a [routerLink]=\"nav.link\">\n        <i [class]='nav.icon'></i>\n        {{nav.title}}\n    </a>\n    <ngx-sidebar *ngIf=\"nav.children\" [navigation]='nav.children' [containerClass]=\"'child-ul'\"></ngx-sidebar>\n  </li>\n</ul>",
                styles: [".main-ul{list-style:none;padding:0;position:relative;margin-top:2px}ul li{display:none;position:relative}ul li a{position:relative}ul li>a:after{content:\"\\e99a\";font-family:material;display:inline-block;top:16px;right:15px;position:absolute;font-size:23px;left:auto;-webkit-transition:-webkit-transform .3s;transition:-webkit-transform .3s;transition:transform .3s;transition:transform .3s,-webkit-transform .3s}ul li.opened>a:after{-webkit-transform:rotate(90deg);transform:rotate(90deg)}ul li>a:only-child:after{content:\"\";border:0}.main-ul>li{display:block}.main-ul>li.opened{background:#f7f7f7;border-left:3px solid #4c84ff}.main-ul>li.opened .child-ul{background:#f9f9f9}.main-ul>li>a:hover{background:#f7f7f7!important}.main-ul li i[class^=icon-]{font-size:22px;padding-right:5px;text-align:center;color:#aaa}.main-ul li a{display:block;font-weight:400;padding:16px 20px;color:#717171;text-decoration:none}.main-ul li.opened>a{color:#4c84ff!important}.main-ul li.opened>ngx-sidebar>.child-ul>li{display:block!important}.child-ul{padding-left:10px;display:none}.main-ul li.opened>ngx-sidebar>.child-ul{display:block!important}.child-ul>li{position:relative}.child-ul>li:hover>a{color:#4c84ff}"]
            },] },
];
NgxSidebarComponent.ctorParameters = function () { return [
    { type: core.Renderer, },
    { type: router.Router, },
    { type: router.ActivatedRoute, },
]; };
NgxSidebarComponent.propDecorators = {
    "navigation": [{ type: core.Input, args: ['navigation',] },],
    "containerClass": [{ type: core.Input, args: ['containerClass',] },],
};
var NgxSidebarModule = /** @class */ (function () {
    function NgxSidebarModule() {
    }
    return NgxSidebarModule;
}());
NgxSidebarModule.decorators = [
    { type: core.NgModule, args: [{
                declarations: [
                    NgxSidebarComponent
                ],
                imports: [
                    platformBrowser.BrowserModule,
                    router.RouterModule.forRoot([])
                ],
                providers: [],
                exports: [
                    NgxSidebarComponent
                ]
            },] },
];
NgxSidebarModule.ctorParameters = function () { return []; };

exports.NgxSidebarComponent = NgxSidebarComponent;
exports.NgxSidebarModule = NgxSidebarModule;

Object.defineProperty(exports, '__esModule', { value: true });

})));
